mkosi/__init__.py

```diff
@@ -934,6 +934,7 @@ def build_initrd(state: MkosiState) -> Path:
         "--make-initrd", "yes",
         "--bootable", "no",
         "--manifest-format", "",
+        *(f"--environment={k}={v}" for k, v in state.config.environment.items()),
```

Comment:
I think we should only plumb `SOURCE_DATE_EPOCH` through, instead of the whole environment.